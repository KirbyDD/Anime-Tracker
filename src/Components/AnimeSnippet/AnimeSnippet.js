import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./AnimeSnippet.scss";

export class AnimeSnippet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: false
    };
  }

  handleEnter = () => {
    this.setState({ title: true });
  };

  handleLeave = () => {
    this.setState({ title: false });
  };

  titleToDisplay = () => {
    if (this.props.enTitle) {
      return this.props.enTitle;
    } else {
      return this.props.jpTitle;
    }
  };

  checkIsFav = () => {
    if (this.props.isFav) {
      return `/watchlist/${this.props.id}`;
    } else {
      return `/anime/${this.props.id}`;
    }
  };

  render() {
    let title = this.titleToDisplay();
    let route = this.checkIsFav();
    return (
      <Link to={route}>
        {this.state.title && <p className="snippet-title">{title}</p>}
        <img
          src={this.props.posterImage.small}
          alt={`poster for ${this.props.enTitle}`}
          className="snippet-image"
          onMouseEnter={e => this.handleEnter()}
          onMouseLeave={e => this.handleLeave()}
        />
      </Link>
    );
  }
}

export default AnimeSnippet;
