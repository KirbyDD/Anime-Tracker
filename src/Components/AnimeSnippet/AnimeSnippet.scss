.snippet-image {
  border: solid 1px;
  box-shadow: 5px 10px 5px gold;
  color: gold;
  height: 400px;
  margin: 5px 12px;
  padding: 7px;
  &:hover {
    box-shadow: 5px 10px 5px rgb(238, 139, 27);
  }
}

.snippet-title {
  background-color: black;
  color: gold;
  font-size: 30px;
  position: absolute;
  text-align: center;
  text-decoration: none;
  z-index: 1;
}