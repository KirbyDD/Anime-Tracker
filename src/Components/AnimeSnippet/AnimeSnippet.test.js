import React from "react";
import { shallow } from "enzyme";
import { AnimeSnippet } from "./AnimeSnippet";

let props = {
  id: 1,
  enTitle: "Life of TronKat",
  jpTitle: "Tronkat no Life desu",
  posterImage: {
    small: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  }
};
let props2 = {
  id: 1,
  enTitle: null,
  jpTitle: "Tronkat no Life desu",
  posterImage: {
    small: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  }
};

describe("Anime Snippet", () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<AnimeSnippet {...props} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot when it isFav is true", () => {
    wrapper = shallow(<AnimeSnippet {...props} isFav={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    wrapper.setState({ title: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default state property", () => {
    expect(wrapper.state()).toEqual({ title: false });
  });

  it("should change state when handleEnter is called", () => {
    wrapper.instance().handleEnter();
    expect(wrapper.state()).toEqual({ title: true });
  });

  it("should change state when handleLeave is called", () => {
    wrapper.setState({ title: true });
    wrapper.instance().handleLeave();
    expect(wrapper.state()).toEqual({ title: false });
  });

  it("should return english title from TitleToDisplay", () => {
    expect(wrapper.instance().titleToDisplay()).toEqual("Life of TronKat");
  });

  it("should return japanese title from TitleToDisplay", () => {
    wrapper = shallow(<AnimeSnippet {...props2} />);
    expect(wrapper.instance().titleToDisplay()).toEqual("Tronkat no Life desu");
  });

  it("should return anime/id url from checkIsFav", () => {
    expect(wrapper.instance().checkIsFav()).toEqual("/anime/1");
  });

  it("should return watchlist/id url from checkIsFav", () => {
    wrapper = shallow(<AnimeSnippet {...props} isFav={true} />);
    expect(wrapper.instance().checkIsFav()).toEqual("/watchlist/1");
  });
});
