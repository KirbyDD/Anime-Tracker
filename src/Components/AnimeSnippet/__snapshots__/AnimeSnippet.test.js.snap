// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Snippet should match snapshot 1`] = `
<Link
  to="/anime/1"
>
  <img
    alt="poster for Life of TronKat"
    className="snippet-image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="https://i.imgur.com/vihivug.jpg"
  />
</Link>
`;

exports[`Anime Snippet should match snapshot 2`] = `
<Link
  to="/anime/1"
>
  <p
    className="snippet-title"
  >
    Life of TronKat
  </p>
  <img
    alt="poster for Life of TronKat"
    className="snippet-image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="https://i.imgur.com/vihivug.jpg"
  />
</Link>
`;

exports[`Anime Snippet should match snapshot when it isFav is true 1`] = `
<Link
  to="/watchlist/1"
>
  <img
    alt="poster for Life of TronKat"
    className="snippet-image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="https://i.imgur.com/vihivug.jpg"
  />
</Link>
`;
